x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/config/Job.java

```diff
@@ -1207,6 +1207,7 @@ public void validateModelSnapshotRetentionSettingsAndSetDefaults() {
                 && modelSnapshotRetentionDays > DEFAULT_DAILY_MODEL_SNAPSHOT_RETENTION_AFTER_DAYS) {
                 dailyModelSnapshotRetentionAfterDays = DEFAULT_DAILY_MODEL_SNAPSHOT_RETENTION_AFTER_DAYS;
             }
+            analysisConfig.setDefaultModelPruneWindowIfNoneProvided();
```

Comment:
Since `analysisConfig` is immutable, this should be changed as follows:

1. Check if the current `analysisConfig` has a model prune window
2. If it doesn't, construct an `AnalysisConfig.Builder` from `analysisConfig`
3. Call `setModelPruneWindow` on that `AnalysisConfig.Builder`
4. Call `setAnalysisConfig` on the `Job.Builder` that this method relates to, passing the `AnalysisConfig.Builder` - this will replace the old `analysisConfig` on the `Job.Builder` with the adjusted one